packages/core/database/lib/index.d.ts

```diff
@@ -2,16 +2,37 @@ import { LifecycleProvider } from './lifecycles';
 import { MigrationProvider } from './migrations';
 import { SchemaProvideer } from './schema';
 
-type BooleanWhere<T> = {
+type LogicOperators<T> = {
   $and?: WhereParams<T>[];
   $or?: WhereParams<T>[];
   $not?: WhereParams<T>;
-};
+}
+
+type AttributeOperators<T, K extends keyof T> = {
+  $eq?: T[K];
+  $ne?: T[K];
+  $in?: T[K][];
+  $notIn?: T[K][];
+  $lt?: T[K];
+  $lte?: T[K];
+  $gt?: T[K];
+  $gte?: T[K];
+  $between?: [T[K], T[K]];
+  $contains?: T[K];
+  $notContains?: T[K];
+  $containsi?: T[K];
+  $notContainsi?: T[K];
+  $startsWith?: T[K];
+  $endsWith?: T[K];
+  $null?: boolean;
+  $notNull?: boolean;
+  $not?: WhereParams<T> | AttributeOperators<T, K>;
+}
 
-type WhereParams<T> = {
-  [K in keyof T]?: T[K];
+export type WhereParams<T> = {
+  [K in keyof T]?: T[K] | T[K][] | AttributeOperators<T, K>;
```

Comment:
I'm not sure to understand the need for `T[K][]` here, could you explain what is its purpose?

Reply:
It's because the [`$in`](https://docs.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/query-engine/filtering.html#in) attribute can be omitted. 
For example, if the data type is a string for a certain attribute (called position in this case):
```json
{
 "where": {
   "position": ["manager", "intern"]
 }
}
```
this would be correct and converted to an `$in` attribute. Therefore the type `T[K][]` has to be supported.

Reply:
Make sense, thanks :slightly_smiling_face: